moniker_src/data_moniker_service.cpp

```diff
@@ -81,218 +96,201 @@ namespace ni
-        {
-            while (true)
+                break;
+            }
+            if (writeRequest->cancel())
             {
-                x = 0;
-                SidebandReadResponse readResult;
-                for (auto reader: readers)
-                {
-                    auto readValue = readResult.mutable_values()->mutable_values(x++);
-                    std::get<0>(reader)(std::get<1>(reader), *readValue);
-                }
-                if (!WriteSidebandMessage(sidebandToken, readResult))
+                std::cout << "Cancelled." << std::endl;
+                break;
+            }
+            if (writers->size() > 0)
+            {
+                int x = 0;                
+                if (writers->size() != writeRequest->values().values_size())
                 {
+                    std::cout << "ERROR: The number of writers and values read did not match" << std::endl;
+                    std::cout << "Expected: " << writers->size() << std::endl;
+                    std::cout << "Actual: " << writeRequest->values().values_size() << std::endl;
                     break;
                 }
-                if (writers.size() > 0)
+                for (auto writer: *writers)
                 {
-                    int x = 0;
-                    if (!ReadSidebandMessage(sidebandToken, &writeRequest))
-                    {
-                        break;
-                    }
-                    if (writeRequest.cancel())
-                    {
-                        break;
-                    }
-                    for (auto writer: writers)
-                    {
-                        std::get<0>(writer)(std::get<1>(writer), const_cast<google::protobuf::Any&>(writeRequest.values().values(x++)));
-                    }
+                    std::get<0>(writer)(std::get<1>(writer), arena, const_cast<google::protobuf::Any&>(writeRequest->values().values(x++)));
                 }
             }
+            auto readResult = Arena::CreateMessage<SidebandReadResponse>(&arena);
+            if (readResult->GetArena() != &arena)
+            {
+                std::cout << "No Arena 111" << std::endl;
+            }
+            if (readers->size() > 0)
+            {
+                 int x = 0;
+                 for (auto reader: *readers)
+                 {
+                     auto readValue = readResult->mutable_values()->add_values();
+                     if (readValue->GetArena() != &arena)
+                     {
+                         std::cout << "No Arena" << std::endl;
+                     }
+                     std::get<0>(reader)(std::get<1>(reader), arena, *readValue);
+                 }
+            }
+            if (!WriteSidebandMessage(sidebandToken, *readResult))
+            {
+                std::cout << "Failed to write" << std::endl;
+                break;
+            }
+            arena.Reset();
+    		std::this_thread::sleep_for(std::chrono::microseconds(10));
```

Comment:
I do not know if there was a usecase for sleeping for the prototype. But since server should run as fast as possible, we should remove it unless it was done throttle the server Read/write loop. I will remove it from the code.